node/nodeTransactions.go

```diff
@@ -150,107 +170,106 @@ func (n *Node) isTxInStorage(hash []byte) bool {
 	if err == nil {
-		return txBytes, unsignedTx, true
+		return txBytes, transaction.TxTypeUnsigned, true
 	}
 
-	return nil, invalidTx, false
+	return nil, transaction.TxTypeInvalid, false
 }
 
-func (n *Node) castObjToTransaction(txObj interface{}, txType transactionType) (*transaction.ApiTransactionResult, error) {
+func (n *Node) castObjToTransaction(txObj interface{}, txType transaction.TxType) (*transaction.ApiTransactionResult, error) {
 	switch txType {
-	case normalTx:
+	case transaction.TxTypeNormal:
+		fallthrough
+	case transaction.TxTypeInvalid:
 		if tx, ok := txObj.(*transaction.Transaction); ok {
-			status := n.computeTransactionStatus(tx, true)
-			return n.prepareNormalTx(tx, status)
+			return n.prepareNormalTx(tx)
 		}
-	case rewardTx:
+	case transaction.TxTypeReward:
 		if tx, ok := txObj.(*rewardTxData.RewardTx); ok {
-			status := n.computeTransactionStatus(tx, true)
-			return n.prepareRewardTx(tx, status)
+			return n.prepareRewardTx(tx)
 		}
-	case unsignedTx:
+	case transaction.TxTypeUnsigned:
 		if tx, ok := txObj.(*smartContractResult.SmartContractResult); ok {
-			status := n.computeTransactionStatus(tx, true)
-			return n.prepareUnsignedTx(tx, status)
+			return n.prepareUnsignedTx(tx)
 		}
 	}
 
-	return &transaction.ApiTransactionResult{Type: string(invalidTx)}, nil // this shouldn't happen
+	return &transaction.ApiTransactionResult{Type: string(transaction.TxTypeInvalid)}, nil // this shouldn't happen
 }
 
-func (n *Node) unmarshalTransaction(txBytes []byte, txType transactionType) (*transaction.ApiTransactionResult, error) {
+func (n *Node) unmarshalTransaction(txBytes []byte, txType transaction.TxType) (*transaction.ApiTransactionResult, error) {
 	switch txType {
-	case normalTx:
+	case transaction.TxTypeNormal:
+		fallthrough
+	case transaction.TxTypeInvalid:
 		var tx transaction.Transaction
 		err := n.internalMarshalizer.Unmarshal(&tx, txBytes)
 		if err != nil {
 			return nil, err
 		}
-		status := n.computeTransactionStatus(&tx, false)
-		return n.prepareNormalTx(&tx, status)
-	case rewardTx:
+		return n.prepareNormalTx(&tx)
+	case transaction.TxTypeReward:
 		var tx rewardTxData.RewardTx
 		err := n.internalMarshalizer.Unmarshal(&tx, txBytes)
 		if err != nil {
 			return nil, err
 		}
-		status := n.computeTransactionStatus(&tx, false)
-		return n.prepareRewardTx(&tx, status)
+		return n.prepareRewardTx(&tx)
 
-	case unsignedTx:
+	case transaction.TxTypeUnsigned:
 		var tx smartContractResult.SmartContractResult
 		err := n.internalMarshalizer.Unmarshal(&tx, txBytes)
 		if err != nil {
 			return nil, err
 		}
-		status := n.computeTransactionStatus(&tx, false)
-		return n.prepareUnsignedTx(&tx, status)
-	default:
-		return &transaction.ApiTransactionResult{Type: string(invalidTx)}, nil // this shouldn't happen
+		return n.prepareUnsignedTx(&tx)
 	}
+
+	return &transaction.ApiTransactionResult{Type: string(transaction.TxTypeInvalid)}, nil // this shouldn't happen
```

Comment:
instead of a comment ca we add a warn here?

Reply:
Fixed.